### ISSUE TITLE
Compose function probability bug

### ISSUE THREAD

**Issue by Edenzzzz** [albumentations-team/albumentations#960]
## 🐛 Bug

**When p1=0,many of the images are produced without any transformation,while they should at least go through one of the 
operations in OneOf function**.When p1=1,the code works as intended and every image is transformed.You can try and run 
the code with p1 explicitly set to 0 and see if that happens.When p1=1,it works just fine.I have no idea about this buggy 
behavior.
>

## To Reproduce

I used the Compose function to transform RGB images with a for loop.
import numpy as np,cv2 as cv,os
os.environ['CUDA_VISIBLE_DEVICES']='-1'
import albumentations as a
import matplotlib.pyplot as plt
from keras.preprocessing.image import ImageDataGenerator
from albumentations import (
    HorizontalFlip, IAAPerspective, ShiftScaleRotate, CLAHE, RandomRotate90,
    Transpose, ShiftScaleRotate, Blur, OpticalDistortion, GridDistortion, HueSaturationValue,
    IAAAdditiveGaussianNoise, GaussNoise, MotionBlur, MedianBlur, IAAPiecewiseAffine,
    IAASharpen, IAAEmboss, VerticalFlip,RandomBrightnessContrast, Flip, OneOf, Compose
)
from keras.datasets import cifar10
from keras.utils import np_utils
num_images=30000
#get data
(x_train, y_train), (x_test, y_test) = cifar10.load_data()
y_train = np_utils.to_categorical(y_train, 10)

x_aug=np.zeros(shape=x_train.shape)
datagen = ImageDataGenerator()
datagen.fit(x_train)
data_iter = datagen.flow(x_train, y_train, batch_size=num_images,shuffle=True)
for x_batch,y_batch in data_iter:
    x_train=x_batch
    y_train=y_batch
    break

p1=round(np.random.rand())
#here is the bug
pipeline=Compose([
                  OneOf([HorizontalFlip(),VerticalFlip()],p=1-p1),
                  HorizontalFlip(p1),
                  VerticalFlip(p1)
                 ])
for i in range(num_images):
    x_aug[i]=pipeline(image=x_train[i])['image']

x_train=x_train.astype('uint8')
x_aug=x_aug.astype('uint8')
y_aug=y_train[:num_images]

for i in range(10):
    plt.subplot(2,5,i+1)
    plt.imshow(x_train[i].astype('uint8'))
plt.title('original')
plt.axis('off')
plt.show()
for i in range(10):
    plt.subplot(2,5,i+1)
    plt.imshow(x_aug[i].astype('uint8'))
plt.title('augmented')
plt.axis('off')
plt.show()
count=0
for i in range(num_images):
    if((x_aug[i]==x_train[i]).all()):
        print('Transform Unsuccessful')
        count+=1
print('\n%d images unsuccessful'%count)
print('p1=',p1)

## Expected behavior
This code is intended to apply one of the three operations on every image:horizontal flip,vertical flip or both operations.

## Environment

 - Albumentations version (e.g., 0.1.8):1.1.0
 - Python version (e.g., 3.7):3.7
 - OS (e.g., Linux):win10
 - How you installed albumentations (`conda`, `pip`, source):pip
 - Any other relevant information:

**Comment by Dipet** [comment]
You have this pipeline:

```python
import random
import albumentations as A

p1 = random.random()
pipeline = A.Compose(
    [
        A.OneOf([A.HorizontalFlip(), A.VerticalFlip()], p=1 - p1),
        A.HorizontalFlip(p1),
        A.VerticalFlip(p1)
    ]
)
```

This way, each transform will be sampled independently, except for transforms inside `OneOf`. 
Why do you think that at every step you will be sure to transform the image? ?

**Comment by Edenzzzz** [comment]
> You have this pipeline:
> 
> ```python
> import random
> import albumentations as A
> 
> p1 = random.random()
> pipeline = A.Compose(
>     [
>         A.OneOf([A.HorizontalFlip(), A.VerticalFlip()], p=1 - p1),
>         A.HorizontalFlip(p1),
>         A.VerticalFlip(p1)
>     ]
> )
> ```
> 
> This way, each transform will be sampled independently, except for transforms inside `OneOf`.
> Why do you think that at every step you will have transformed image?
> 
> Another problem, that you may apply `HorizontalFlip` inside `OneOf` and after that do the same transform and you will have as a result original image.

Thanks alot,that turns out to be a silly mistake......
To prevent the second problem I rounded p1 so that it can only be 0 or 1

**Comment by Dipet** [comment]
When `p1=1` `OneOf` block will never be applied. And you will have an flipped image horizontally and vertically. 
When `p1=0` only one flip from `OneOf` will be sampled and applied.
